flexxamples/howtos/openlayers.py

```diff
@@ -0,0 +1,177 @@
+
+    @flx.action
+    def remove_drawing_interaction(self):
+        self.map.removeInteraction(self.drawPoint)
+        self.map.removeLayer(self.drawVectorLayer)
+        self.drawVectorLayer.getSource().clear()
+
+    @flx.action
+    def map_init(self):
+        global ol
+        self.olview = ol.View({
+            "zoom": 8,
+            "center": [-80.901813, 22.968599],
+            "projection": "EPSG:4326",
+            "minZoom": 3,
+            "maxZoom": 100
+        })
+
+        self.baseLayer = ol.layer.Tile({
+            "source": ol.source.OSM(),
+        })
+
+        self.vectorLayer = ol.layer.Vector({
+            "source": ol.source.Vector({
+                "format": ol.format.GeoJSON()
+            }),
+            "name": "Vector",
+            "style": ol.style.Style({
+                "image": ol.style.Circle({
+                    "radius": 7,
+                    "fill": ol.style.Fill({
+                        "color": 'rgba(255, 0, 0, 0.5)'
+                    })
+                })
+            }),
+        })
+
+        self.drawVectorLayer = ol.layer.Vector({
+            "source": ol.source.Vector({
+                "format": ol.format.GeoJSON()
+            }),
+            "name": "Draw Vector",
+            "style": ol.style.Style({
+                "image": ol.style.Circle({
+                    "radius": 7,
+                    "fill": ol.style.Fill({
+                        "color": 'rgba(0, 255, 0, 0.5)'
+                    })
+                })
+            }),
+        })
+
+        self.drawPoint = ol.interaction.Draw({
+            "type": 'Point',
+            "source": self.drawVectorLayer.getSource()
+        })
+
+        self.map_config = {
+            "target": self.mapnode,
+            'view': self.olview,
+            "controls": [ol.control.Zoom(), ol.control.MousePosition()],
+            "layers": []
+        }
+        self.map = ol.Map(self.map_config)
+        self.map.on('click', self.pointer_event)
+
+    @flx.emitter
+    def pointer_event(self, event):
+        return {"event": event}
+
+    @flx.action
+    def add_vector_layer(self):
+        format = self.vectorLayer.getSource().getFormat()
+        features = format.readFeatures(geojson)
+        self.vectorLayer.getSource().clear()
+        self.vectorLayer.getSource().addFeatures(features)
+        self.map.addLayer(self.vectorLayer)
+
+    @flx.action
+    def add_osm_layers(self):
+        self.map.addLayer(self.baseLayer)
+
+    def _create_dom(self):
+        """ This function gets automatically called when needed; Flexx is aware
+        of what properties are used here.
+        """
+        global document
+        node = document.createElement('div')
+        self.mapnode = document.createElement('div')
+        node.appendChild(self.mapnode)
+        self.mapnode.id = 'maproot'
+        return node
+
+    def _render_dom(self):
+        self.map_init()
+        return super()._render_dom()
```

Comment:
I wonder if you want to call `map_init()` here. The name suggests that you want to call it just once :)

Reply:
Yes,  You think I should do it on _create_dom?

Reply:
Thanks for all the useful comments.

Reply:
> You think I should do it on _create_dom?

Yes, probably.

Reply:
I think this is the last remaining issue right now, right?

Reply:
yes, I couldn't make it work inside _create_dom. I putted an if inside map_init so it wont be called every time. Is there something like an after render event?
```suggestion
        if not self.initialised:
```

Reply:
Maybe put it in `init()`? That's usually a good point to initialize stuff, since the DOM and properties have been initialized.

Reply:
Maybe put it in `init()`? That's usually a good point to initialize stuff, since the DOM and properties have been initialized.

Reply:
Maybe put it in `init()`? That's usually a good point to initialize stuff, since the DOM and properties have been initialized.

Reply:
I've try on the init() without luck. I need to call map_init() after the div that contains the map is loaded on the page. Normally this is done by putting your code after the div. Any Ideas?

Reply:
I also was thinking about what you said about kipping the repo small  by using CDN. I could use CDN for this example and create another small example demostrating how to load local assets but with smaller css, js files. what do you think?

Reply:
Mmm odd that it does not work in `init()`. Ah well, I guess we can leave it as-is then :)

> I could use CDN for this example and create another small example demostrating how to load local assets but with smaller css, js files.

That would be even better, yes!

Reply:
ok, I've created a new example and update the last one with CDN.